### DISCUSSION TITLE
Physics Convex Hull Collider Bug

### DISCUSSION THREAD

**Discussion by mathmb1986** [stride3d/stride#1574] [General]
https://github.com/mathmb1986/RigidBodyPhysicBug/blob/master/ColliderBug.png
https://github.com/mathmb1986/RigidBodyPhysicBug

 
there is no crash.. the collision is far from corresponding to the object

What I want to do is.. that the pieces explode and fall to the ground but if I use a staticphysicsComponent.. with a static mesh .. Made with a model asset the collision with a bullet And all object is perfect but.. the object does not move ..so we take RigidBodyphysicsComponent ... if I take a static mesh .. Made with a model asset there is no collision of Detect and that even if the groups and Flags are put to All .. 

if I use RigidBodyPhysicsComponent with a Convex model made with a model asset..  the Object can fall... The form in the debug is the right ... but there is far too much space between the colliding Objects

**Comment by manio143** [comment]
@mathmb1986 If there's an issue with Stride could you please describe the bug in more detail in an issue? (i.e file a proper bug report in the "issues" tab)
If this is an issue in your code and you're looking for help fixing it, please provide more context here.

**Reply by mathmb1986** [reply]
I Found an issue ..  Add the Margin to ConvexHullShape

https://github.com/mathmb1986/RigidBodyPhysicBug/blob/master/ColliderShapeAssetCompiler.cs
https://github.com/mathmb1986/RigidBodyPhysicBug/blob/master/ConvexHullColliderShape.cs
https://github.com/mathmb1986/RigidBodyPhysicBug/blob/master/ConvexHullColliderShapeDesc.cs

**Reply by manio143** [reply]
Could you please create an issue and describe in more details what is the problem in the code?
Or is that an issue in your code specifically?

**Reply by mathmb1986** [reply]
In my code no problem .  missing parameter in StridePhysicsEngine missing the Margin parameter ...
what is margin  https://www.youtube.com/watch?v=BGAwRKPlpCw   

https://github.com/mathmb1986/RigidBodyPhysicBug/blob/master/ColliderShapeAssetCompiler.cs
https://github.com/mathmb1986/RigidBodyPhysicBug/blob/master/ConvexHullColliderShape.cs
https://github.com/mathmb1986/RigidBodyPhysicBug/blob/master/ConvexHullColliderShapeDesc.cs

Updated Version of StrideEngine
